src/racer/core.rs

```diff
@@ -97,6 +97,15 @@ pub struct Match {
     pub docs: String,
 }
 
+impl Match {
+    /// Checks if two matches can be considered the same for deduplication purposes.
+    fn is_same_as(&self, other: &Match) -> bool {
```

Comment:
I didn't want to publicly expose the trait. If we ever had to remove it in the future, it'd be a breaking change.